src/llama_recipes/finetuning.py

```diff
@@ -151,11 +151,16 @@ def main(**kwargs):
         model.to(torch.bfloat16)
 
     if train_config.use_peft:
-        peft_config = generate_peft_config(train_config, kwargs)
-        model = get_peft_model(model, peft_config)
+        # Load the pre-trained peft model checkpoint and setup its configuration
+        if train_config.from_peft_checkpoint:
+            model = PeftModel.from_pretrained(model, train_config.from_peft_checkpoint, is_trainable=True)
+        # Generate the peft config and start fine-tuning from original model
+        else:
+            peft_config = generate_peft_config(train_config, kwargs)
+            model = get_peft_model(model, peft_config)
+            if wandb_run:
+                wandb_run.config.update(peft_config)
```

Comment:
This should also run if we resume from a checkpoint. You can retrieve the peft_config from the peft_model in both cases https://github.com/huggingface/peft/blob/0649947396a946a9333b731bea2e76551e85ad92/src/peft/peft_model.py#L151